### PR TITLE
bugfix/support optional access configs on connection configs

### PR DESCRIPTION
**Pull request by rbiseck3**
### Description
Having a connection config implemented with the access config typing set to `Optional` breaks the mapping from flat data. This updates that code to handle it.

### CODE REVIEW

File: test/unit/cli/test_utils_v2.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+from unstructured_ingest.v2.interfaces import AccessConfig, ConnectionConfig
+
+
+def test_extract_config_optional_access_config():
+    class MyAccessConfig(AccessConfig):
+        host: str
+
+    class MyModel(ConnectionConfig):
+        v: int
+        access_config: Optional[Secret[MyAccessConfig]]
```
Comment (potter-potter) [inline test/unit/cli/test_utils_v2.py:17]: I was expecting the Optional to be in line 13 if we had multiple AccessConfigs and some are Optional.

Comment (rbiseck3) [reply test/unit/cli/test_utils_v2.py:17]: There should only ever be one access config

Comment (rbiseck3) [reply test/unit/cli/test_utils_v2.py:17]: We already have plenty of access configs that have optional fields and that works fine.

Comment (potter-potter) [reply test/unit/cli/test_utils_v2.py:17]: I meant multiple optional fields in the AccessConfigs.  But sounds like it works fine already.

Comment (rbiseck3) [reply test/unit/cli/test_utils_v2.py:17]: Example of one that already exists that works: [milvus.py](https://github.com/Unstructured-IO/unstructured-ingest/blob/main/unstructured_ingest/v2/processes/connectors/milvus.py#L34)

File: unstructured_ingest/v2/cli/utils/click.py [hunk 2]
```diff
@@ -112,13 +112,28 @@ def convert(
 BaseModelT = TypeVar("BaseModelT", bound=BaseModel)
 
 
+def unwrap_optional(val: Any) -> tuple[Any, bool]:
```
Comment (potter-potter) [inline unstructured_ingest/v2/cli/utils/click.py:115]: This definitely makes my eyes water.

**Review by potter-potter — APPROVED** [review]
lgtm

**Pull request merged**